### DISCUSSION TITLE
Single Project Style Structure In Uno-Platform

### DISCUSSION THREAD

**Discussion by hintsofttech** [unoplatform/uno#7339] [Ideas]
WHEN UNO-FLATFORM WILL BE SINGLE PROJECT TARGETTING MULTIPLE PLATFORMS ? 

LIKE .NET MAUI IS ABOUT TO DO...

WE ARE EXTREMELY WAITING UNO-PLATFORM ANNOUNCEMENTS TO THIS DIRECTION...

**Comment by jeromelaban** [comment]
Single project will not be supported because some of the Uno supported .NET targets cannot support this (WPF, Gtk and WebAssembly). 

You may notice that we have already reduced the number of projects in the VS2022 template when targeting .NET 6, where Android, iOS, macOS and Catalyst are now in the same project.

**Reply by jeromelaban** [reply]
Uno Platform 5.2 brings single project for all targets.

**Comment by robloo** [comment]
Duplicate of https://github.com/unoplatform/uno/discussions/6096

**Comment by HavenDV** [comment]
At the moment, you can try the single project solution within a separate repo - https://github.com/HavenDV/Uno.Sdk. This is a workload similar to tizen and this delivers`net7.0-webassembly;net7.0-gtk;net7.0-linux` to distinguish WebAssembly/Skia.Gtk/Skia.Linux.Framebuffer respectively. Skia.Wpf is implemented within `net7.0-windows` (this is an implicit indication for `net7.0-windows7`, which is different from the WinUI TargetFramework - `net7.0-windows10.x.x.x`).
Check the documentation for more details.
I would be glad to receive any feedback

**Comment by xperiandri** [comment]
Should this be closed as resolved?
